File: src/main/java/io/github/smootheez/curseforge/ChangelogType.java
package io.github.smootheez.curseforge;

import com.google.gson.annotations.*;

/**
 * Defines the supported changelog content formats used by curseforge.
 */
public enum ChangelogType {

    /** Changelog provided as plain text. */
    @SerializedName("text")
    TEXT,

    /** Changelog written using Markdown syntax. */
    @SerializedName("markdown")
    MARKDOWN,

    /** Changelog provided as HTML content. */
    @SerializedName("html")
    HTML
}

File: src/main/kotlin/io/github/smootheez/curseforge/ChangelogType.kt
package io.github.smootheez.curseforge

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
enum class ChangelogType {
    @SerialName("text")
    TEXT,

    @SerialName("markdown")
    MARKDOWN,

    @SerialName("html")
    HTML
}